sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -363,6 +363,18 @@ private[sql] trait SQLTestUtilsBase
     }
   }
 
+  /**
+   * Drops namespace `namespace` after calling `f`.
+   */
+  protected def withNamespace(namespaces: String*)(f: => Unit): Unit = {
+    Utils.tryWithSafeFinally(f) {
+      namespaces.foreach { name =>
+        spark.sql(s"DROP NAMESPACE IF EXISTS $name CASCADE")
+      }
+      spark.sql(s"USE default")
```

Comment:
We should set back to the default namespace of the current catalog. Not sure we have such a command yet, cc @imback82

Reply:
We can get current catalog and set back like this code. 
```
spark.sessionState.catalogManager.currentCatalog match {
        case supportNamespace: SupportsNamespaces =>
          supportNamespace.asNamespaceCatalog.defaultNamespace() match {
            case Array(name) => spark.sql(s"USE $name")
            case Array(name, _*) => spark.sql(s"USE $name")
            case _ => spark.sql("USE default")
          }
        case _ => spark.sql("USE default")
      }
```

Reply:
Let's remove it first, and assume the test code won't switch the current namespace.

Things can get complicated if the test code switches the current catalog.

Reply:
Shall we add some doc to remind like this ?
```
If you will change current catalog, you should use this method carefully. This may be remove the wrong namespace.
```

Reply:
how about we add doc to remind callers to not switch current namespace?

Reply:
Updated, please take an another review.

Reply:
> We should set back to the default namespace of the current catalog. Not sure we have such a command yet, cc @imback82

No, we don't have a such a command yet. Do you think it will be useful to have one?

This also reminds me about the behavior of dropping namespace. If you are trying to drop a namespace, which is set to the current namespace, this shouldn't be allowed right? We don't have a such a check yet, but I can add one if needed.